md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDAO.java

```diff
@@ -702,5 +726,97 @@ else  if (childProcess.getProcessType().getProcessTypeId() == 36){
+                    )
+            {return "ACCESS GRANTED";}
+            else
+            {LOGGER.info("user read");throw new SecurityException("ACCESS DENIED");}
+            case "execute": if (executeList.contains(process.getPermissionTypeByUserAccessId().getPermissionTypeId())||
+                    (userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                    || (!userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+                    )
+            {return "ACCESS GRANTED";}
+            else
+            {LOGGER.info("user execute");throw new SecurityException("ACCESS DENIED");}
+        }
+        {LOGGER.info("no user");throw new SecurityException("ACCESS DENIED");}    }
+    else{
+    if (userRolesNameList.contains(processCreater))
+    {
+        switch (action){
+            case "write": if (writeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                               {return "ACCESS GRANTED";}
+                           else
+                          {LOGGER.info("group write");throw new SecurityException("ACCESS DENIED");}
+            case "read": if (readList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                               {return "ACCESS GRANTED";}
+                           else
+                          {LOGGER.info("group read");throw new SecurityException("ACCESS DENIED");}
+            case "execute": if (executeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                            {return "ACCESS GRANTED";}
+                            else
+                            {LOGGER.info("group execute");throw new SecurityException("ACCESS DENIED");}
+        }
+        {LOGGER.info("group");throw new SecurityException("ACCESS DENIED");}
+    }
+    else {
+        switch (action){
+            case "write": if (writeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+            {return "ACCESS GRANTED";}
+            else
+            {LOGGER.info("other write");throw new SecurityException("ACCESS DENIED");}
+            case "read": if (readList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+            {return "ACCESS GRANTED";}
+            else
+            {LOGGER.info("other read");throw new SecurityException("ACCESS DENIED");}
+            case "execute": if (executeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+            {return "ACCESS GRANTED";}
+            else
+            {LOGGER.info("other execute");throw new SecurityException("ACCESS DENIED");}
+        }
+        {LOGGER.info("other");throw new SecurityException("ACCESS DENIED");}
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this closing curly brace to the next line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00122)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1199)